internal/jobqueue/jobqueue.go

```diff
@@ -0,0 +1,89 @@
+// Package jobqueue provides a generic interface to a simple job queue.
+//
+// Jobs are pushed to the queue with Enqueue(). Workers call Dequeue() to
+// receive a job and FinishJob() to report one as finished.
+//
+// Each job has a type and arguments corresponding to this type. These are
+// opaque to the job queue, but it mandates that the arguments must be
+// serializable to JSON. Similarly, a job's result has opaque result arguments
+// that are determined by its type.
+//
+// A job can have dependencies. It is not run until all its dependencies have
+// finished successfully. If one of its dependencies fails, it is not run at
```

Comment:
Yeah, good point. I've changed it to be always run, which allowed me to drop quite some code, because a job now doesn't need track success status anymore. Thanks for the suggestion!